### PR TITLE
android: disable network interface monitoring on ANDROID_API >= 30

### PR DESCRIPTION
**Pull request by Danielius1922**

### CODE REVIEW

File: port/android/ipadapter.c [hunk 1]
```diff
@@ -66,6 +66,10 @@
 #include <sys/un.h>
 #include <unistd.h>
 
+#if __ANDROID_API__ < 30
```
Comment (baabel) [inline port/android/ipadapter.c:69]: should this be...  #if __ANDROID_API__ > 29

**Comment by baabel** [comment]
Not necessarily related to this change but....

api/cloud/oc_cloud.c

when OC_NETWORK_MONITOR is not defined, the code in oc_cloud.c calls some undefined functions...(ex: oc_remove_network_interface_event_callback) and they end up as UNDEFINED in the .so library and loadLibrary() fails to load libiotivity-lite-jni.so.  This is because the implementation is conditional in ipadapter.c based on OC_NETWORK_MONITOR.

potential fix:
include the OC_NETWORK_MONITOR directive around calls line #342 and #359
example:
#if defined(OC_SESSION_EVENTS) && defined(OC_NETWORK_MONITOR)
  oc_remove_session_event_callback_v1(cloud_ep_session_event_handler, ctx,
                                      false);
  oc_remove_network_interface_event_callback(cloud_interface_event_handler);
  oc_add_session_event_callback_v1(cloud_ep_session_event_handler, ctx);
  oc_add_network_interface_event_callback(cloud_interface_event_handler);
#endif /* OC_SESSION_EVENTS */

File: port/android/oc_config.h [hunk 2]
```diff
@@ -28,8 +28,13 @@ typedef uint64_t oc_clock_time_t;
 /* Maximum wait time for select function */
 #define SELECT_TIMEOUT_SEC (1)
 
+/* The current implementation in Android port relies on notifications from a
+ * NETLINK socket to retrieve interface change notifications, but this is
+ * restricted from ANDROID_API >= 30  */
+#if __ANDROID_API__ < 30
```
Comment (baabel) [inline port/android/oc_config.h:34]: Should this be defined only for > 29 ?

**Comment by Danielius1922** [comment]
> 

Yeah, the ifdefs are wrong, seems like `OC_SESSION_EVENTS` and `OC_NETWORK_MONITOR` got mixed up, I'll clean it up.

**Comment by baabel** [comment]
@Danielius1922  Hi, I tested latest change. Looks like only 1 more undefined symbol remaining, so when launching android app ...dlopen failed: cannot locate symbol "oc_resource_set_access_in_RFOTM" referenced by...

Also ran nm on the .so library and that seems to be the only undefined symbol in iotivity lib

**Comment by Danielius1922** [comment]
@baabel Should be all good now!

**Comment by baabel** [comment]
thanks.  lgtm, android app launches.

**Comment by Danielius1922** [comment]
Great, I've fired up the Conformance tests, if they all pass I'll merge the fixes to master (they usually take from 3 to 4 hours...)

**Pull request merged**